Feature/remaining controllers

lepeshcoder opened the pull request:
[Данила боев мы тебя умоляем пожалуйста посмотри наш пулл-реквест умоляю тебя]

lepeshcoder committed:
chat-controller

the-man-w-laughs committed:
Add models to store media owners.

lepeshcoder committed:
Chat-controller add methods

the-man-w-laughs committed:
Change order of controllers methods.

the-man-w-laughs committed:
Merge branch 'feature/Chat-controller' into feature/media

lepeshcoder committed:
add some Methods

lepeshcoder committed:
fix

Commit:
Add all repositories, clean controllers

Commit:
refactor chat controller using new repositories

the-man-w-laughs committed:
MediasController setup.

the-man-w-laughs committed:
Update gitignore.

the-man-w-laughs committed:
Merge with refactor/chat-controller.

lepeshcoder committed:
yet another commit

lepeshcoder committed:
add community-controller

the-man-w-laughs committed:
Add initial implementation of medias controller.

the-man-w-laughs committed:
Merge with feature/Users-controller.

the-man-w-laughs committed:
Swagger-tested version of media-controller.

lepeshcoder committed:
fix CommunityController

the-man-w-laughs committed:
Merged with last version of feature/Users-controller.

the-man-w-laughs committed:
Add patch methods.

Commit:
add message-controller

the-man-w-laughs committed:
Merge with feature/message-controller
.

the-man-w-laughs committed:
Add documentation for auth and chat controllers.

the-man-w-laughs committed:
Document community controller.

the-man-w-laughs committed:
Document media controller.

the-man-w-laughs committed:
Fix ProducesResponseType.

lepeshcoder committed:
Merge pull request #19 from lepeshcoder/feature/documentation

Feature/documentation

the-man-w-laughs committed:
Add managed communities and fix some bugs.

lepeshcoder committed:
add Friends-followers functional

lepeshcoder committed:
Communities functional

the-man-w-laughs committed:
Refactor media service.

the-man-w-laughs committed:
Add AthenticationMiddleware.

lepeshcoder committed:
Merge branch 'feature/remaining-controllers' into feature/Friends-followers

lepeshcoder committed:
Merge pull request #21 from lepeshcoder/feature/Friends-followers

Feature/friends followers

lepeshcoder committed:
refactor Auth Controller

the-man-w-laughs committed:
Refactor community controller and change media logic.

lepeshcoder committed:
refactor chat controller

the-man-w-laughs committed:
Refactor communities.

lepeshcoder committed:
Merge branch 'feature/remaining-controllers' into refactor/Auth,Chat

lepeshcoder committed:
Merge pull request #22 from lepeshcoder/refactor/Auth,Chat

Refactor/auth,chat

lepeshcoder committed:
Merge pull request #22 from lepeshcoder/refactor/Auth,Chat

Refactor/auth,chat

lepeshcoder committed:
added GlobalExceptionHandlerMidleware

Commit:
Refactor MessagesController

the-man-w-laughs committed:
Merge with last version of feature/remaining-controllers.

Commit:
remove try-catch handlers in MessagesController

lepeshcoder committed:
Merge remote-tracking branch 'origin/refactor/Messages,Users' into feature/remaining-controllers

# Conflicts:
#	SocialNetwork/SocialNetwork.API/Controllers/MessagesController.cs

boy-eff commented on SocialNetwork/SocialNetwork.API/Controllers/AdminController.cs:
```diff
@@ -0,0 +1,107 @@
+    public async Task<ActionResult<UserResponseDto>> DeleteAdminMedias([FromRoute, Required] uint mediaId)
+    {
+        var deletedMedia = await _mediaService.DeleteMedia(mediaId);
+        if (System.IO.File.Exists(deletedMedia.FilePath))
```
Move logic to BLL

boy-eff commented on SocialNetwork/SocialNetwork.API/Controllers/MediasController.cs:
```diff
@@ -0,0 +1,151 @@
+    {
+        var localMedia = await _mediaService.GetLocalMedia(mediaId);
+
+        if (System.IO.File.Exists(localMedia.FilePath))
```
Move logic to BLL

boy-eff commented on SocialNetwork/SocialNetwork.API/Program.cs:
```diff
@@ -10,9 +14,18 @@
 builder.Services.RegisterBllDependencies(builder.Configuration);
 
+builder.Services.AddControllers().AddNewtonsoftJson(options =>
+    options.SerializerSettings.ReferenceLoopHandling = Newtonsoft.Json.ReferenceLoopHandling.Ignore
+);
+
 // Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
 builder.Services.AddEndpointsApiExplorer();
-builder.Services.AddSwaggerGen();
+builder.Services.AddSwaggerGen(config =>
+{
+    var xmlFile = $"{Assembly.GetExecutingAssembly().GetName().Name}.xml";
+    var xmlPath = Path.Combine(AppContext.BaseDirectory, xmlFile);
+    config.IncludeXmlComments(xmlPath);
+});
```
Move this to extension methods

boy-eff commented on SocialNetwork/SocialNetwork.BLL/Contracts/IChatService.cs:
```diff
@@ -0,0 +1,27 @@
+
+namespace SocialNetwork.BLL.Contracts;
+
+public interface IChatService
```
Use Async postfix for methods, that return Task

boy-eff commented on SocialNetwork/SocialNetwork.BLL/Services/Auth/AuthService.cs:
```diff
@@ -0,0 +1,124 @@
+        return users.FirstOrDefault();
+    }
+
+    public bool IsLoginValid(string login)
```
Use some validation libraries for validation purposes: FluentValidation for example

Commit:
Refactor UsersController

the-man-w-laughs committed:
Split mapper.

the-man-w-laughs committed:
Add CommentsController

the-man-w-laughs committed:
Little refactor Media, Commeents.

the-man-w-laughs committed:
Remake posts db schema.

lepeshcoder committed:
Add admin-controller

the-man-w-laughs committed:
Add PostLikes.

the-man-w-laughs committed:
Merge branch 'feature/remaining-controllers' of https://github.com/lepeshcoder/ModsenPractice into feature/remaining-controllers

Commit:
Refactor AdminController

Commit:
Merge remote-tracking branch 'origin/refactor/admin-controller' into feature/remaining-controllers

Commit:
Refactor AuthController

the-man-w-laughs committed:
Refactor media.

the-man-w-laughs committed:
Merge branch 'feature/remaining-controllers' of https://github.com/lepeshcoder/ModsenPractice into feature/remaining-controllers

the-man-w-laughs committed:
Add post controller templates.

the-man-w-laughs committed:
Refactor ChatService.

the-man-w-laughs committed:
Implement PostService.

the-man-w-laughs committed:
Implement PostService.

Commit:
Refactor ChatController, replace SendMessage to MessageController, setup some AutoMapper fields

the-man-w-laughs committed:
Merge branch 'refactor/chat-controller' of https://github.com/lepeshcoder/ModsenPractice into feature/remaining-controllers

the-man-w-laughs committed:
Fix several moments.

Major changes:
- Add SwaggerExtension and ControllerExtension;
- Implement changing of messages (including mapping);
- Change the order of methods in controllers;
- Implement GET - users/account.

the-man-w-laughs committed:
Merge branch 'feature/remaining-controllers' of https://github.com/lepeshcoder/ModsenPractice into feature/remaining-controllers
